### ISSUE TITLE
Incompatible GigaChannel database format being gossiped

### ISSUE THREAD

**Issue by qstokkink** [Tribler/tribler#4736]
Seems like there is a format inconsistency somewhere:

```python
ERROR   1565602853.14       gigachannel_manager:242  (GigaChannelManager)  Error when processing channel dir download: [Failure instance: Traceback: <class 'Tribler.Core.Modules.MetadataStore.serialization.UnknownBlobTypeException'>: 
/usr/lib/python2.7/threading.py:801:__bootstrap_inner
/usr/lib/python2.7/threading.py:754:run
/home/quinten/.local/lib/python2.7/site-packages/twisted/_threads/_threadworker.py:46:work
/home/quinten/.local/lib/python2.7/site-packages/twisted/_threads/_team.py:190:doWork
--- <exception caught here> ---
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/threadpool.py:250:inContext
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/threadpool.py:266:<lambda>
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/context.py:122:callWithContext
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/context.py:85:callWithContext
/home/quinten/Documents/tribler/Tribler/Core/Modules/gigachannel_manager.py:238:_process_download
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:241:process_channel_dir
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:278:process_mdblob_file
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:287:process_compressed_mdblob
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:307:process_squashed_mdblob
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/serialization.py:71:read_payload_with_offset
]
ERROR   1565602853.14       gigachannel_manager:242  (GigaChannelManager)  Error when processing channel dir download: [Failure instance: Traceback: <class 'Tribler.Core.Modules.MetadataStore.serialization.UnknownBlobTypeException'>: 
/usr/lib/python2.7/threading.py:801:__bootstrap_inner
/usr/lib/python2.7/threading.py:754:run
/home/quinten/.local/lib/python2.7/site-packages/twisted/_threads/_threadworker.py:46:work
/home/quinten/.local/lib/python2.7/site-packages/twisted/_threads/_team.py:190:doWork
--- <exception caught here> ---
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/threadpool.py:250:inContext
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/threadpool.py:266:<lambda>
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/context.py:122:callWithContext
/home/quinten/.local/lib/python2.7/site-packages/twisted/python/context.py:85:callWithContext
/home/quinten/Documents/tribler/Tribler/Core/Modules/gigachannel_manager.py:238:_process_download
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:241:process_channel_dir
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:278:process_mdblob_file
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:287:process_compressed_mdblob
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/store.py:307:process_squashed_mdblob
/home/quinten/Documents/tribler/Tribler/Core/Modules/MetadataStore/serialization.py:71:read_payload_with_offset
]
```

**Comment by qstokkink** [comment]
We should update the GigaChannel community id if the database formats are incompatible.

We decided to do this only just before the official `7.3` release, to fragment the community as little as possible.

**Comment by ichorid** [comment]
The idea was that whatever :hankey: the channel author would put in their channel, it should not crash the clients who download it. If it does not crash for you, that's already fine. However, if it goes into an endless cycle of "trying to process the channel dir", the thing should stop.

We have to decide what to do with the broken channels. Unsubscribe? Blacklist? Wait for an updated version?

**Comment by qstokkink** [comment]
Probably we should just wait for an updated version.

However, if this is the result of simply using previous versions together with the current version we can do with simply updating the community version.

**Comment by qstokkink** [comment]
Closed by #4762. If we want a more long term solution, we should open another issue.
